### ISSUE TITLE
multi processing 

### ISSUE THREAD

**Issue by zeynali** [tensorflow/models#3303]
hi
As you Know , One of the problems in the Train network , speed prediction of Network classification and there is an inconsistency between CPU and GPU. that is , GPU is waiting for CPU to prepares the batch data.
in you opinion , What to do to overcome this problem? i read the inception folder in the research folder , in image_processing.py , there are some of Flag , i thought this Flags are for overcome the issues , please explain these Flags : 

tf.app.flags.DEFINE_integer('batch_size', 32,
                            """Number of images to process in a batch.""")
tf.app.flags.DEFINE_integer('image_size', 299,
                            """Provide square images of this size.""")
tf.app.flags.DEFINE_integer(**'num_preprocess_threads', 4,**
                            """Number of preprocessing threads per tower. """
                            """Please make this a multiple of 4.""")
tf.app.flags.DEFINE_integer(**'num_readers', 4,**
                            """Number of parallel readers during train.""")

# Images are preprocessed asynchronously using multiple threads specified by
# --num_preprocss_threads and the resulting processed images are stored in a
# random shuffling queue. The shuffling queue dequeues --batch_size images
# for processing on a given Inception tower. A larger shuffling queue guarantees
# better mixing across examples within a batch and results in slightly higher
# predictive performance in a trained model. Empirically,
# --input_queue_memory_factor=16 works well. A value of 16 implies a queue size
# of 1024*16 images. Assuming RGB 299x299 images, this implies a queue size of
# 16GB. If the machine is memory limited, then decrease this factor to
# decrease the CPU memory footprint, accordingly.
tf.app.flags.DEFINE_integer**('input_queue_memory_factor', 16,**
                            """Size of the queue of preprocessed images. """
                            """Default is ideal but try smaller values, e.g. """
                            """4, 2 or 1, if host memory is constrained. See """
                            """comments in code for more details.""")

On what principles they set up? are these Flags used in TF-slim library ? i didn't see **input_queue_memory_factor** in TF-Slim library. 

my system Titan x , core i 5 , ram 16G
in your opinion , what are best chose for my system in num_preprocess_threads ?  num_readers ? input_queue_memory_factor ? 
are the batch_size and image_size effective for chose those flags above ?

**Comment by reedwm** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!
